bullet/src/Base.hh

```diff
@@ -248,9 +248,9 @@ class Base : public Implements3d<FeatureList<Feature>>
   public: using CollisionInfoPtr = std::shared_ptr<CollisionInfo>;
   public: using JointInfoPtr  = std::shared_ptr<JointInfo>;
 
-  public: std::unordered_map<std::size_t, WorldInfoPtr> worlds;
-  public: std::unordered_map<std::size_t, ModelInfoPtr> models;
-  public: std::unordered_map<std::size_t, LinkInfoPtr> links;
+  public: std::map<std::size_t, WorldInfoPtr> worlds;
+  public: std::map<std::size_t, ModelInfoPtr> models;
+  public: std::map<std::size_t, LinkInfoPtr> links;
```

Comment:
The reason for this changed didn't jump out at me.  Is there a reason to use `map` vs `unordered_map`?

Reply:
We use the index of the map, I believe `std::unordered_map` is not keeping the order of the data structrue

Reply:
we should return the index here:

https://github.com/gazebosim/gz-physics/blob/f15cb9c51750d68a029b80f1ef7169ae23c7812f/bullet/src/EntityManagementFeatures.cc#L343-L364

we use the index here:

https://github.com/gazebosim/gz-physics/blob/f15cb9c51750d68a029b80f1ef7169ae23c7812f/bullet/src/EntityManagementFeatures.cc#L287-L307